### DISCUSSION TITLE
Documentation for newbie

### DISCUSSION THREAD

**Discussion by goofy2k** [arendst/Tasmota#15800] [Q&A] [answered]
Hi I am a new Tasmota user. I worked through "Getting started" and got something running!

When arriving at "After Configuration" in "Getting Started", following the "controlled" link sends me into the "Commands" section. Yet, I think, it is more important to learn something more on how to set up the device a bit more.......

I have a problem with understanding the differences between / complementary of using Configuration/Configure Template and  Configuration/Configure Template.

Can anyone explain this a bit on a high level? I miss a walkthrough example. The very details can of course be found in the docs.

**Comment by sfromis** [comment]
I assume that you mean the difference between configure module vs template. Seen from the perspective of getting a commercial device, like a plug with energy monitoring, the old style with configure module would either require the user to configure pin by pin, or Tasmota to include the device settings in the binary. With configure template, you can copy/paste one string for that whole part of the setup, much easier and does not take space in the Tasmota for over 2000 known devices.

Conceptually, the template config defines the inherent functions of the hardware. When using a template, only pins configured as "User" are visible in configure module, assumed to be easily accessible in the hardware.

For commercial devices, preferably use configure template. If you do your own tinkering, you can also use configure module, but configure template still makes sense for a "finished" setup, suitable for saving, moving to other similar devices, or putting into your notes.

When using a template, you need to activate it by using module 0.

**Reply by goofy2k** [reply]
OK, this helps!

Some additional questions:

In Configure Module I can select from a list of modules. When I do so I don't see changes in the list of GPIO lines under the Module type entry box. I would expect that predefined settings for that particular module would become visible.

How to use Module 0 ?(your last line)

What is the meaning of the number between () behind each Module?

**Comment by goofy2k** [comment, marked as answer]
I see an "Activate" checkbox  under Configure/Other. Is that what you mean?

Going to change the name in the Template string and see if that item appears in some lists ;-)

**Reply by sfromis** [reply]
Yes, that is one way of activating it.
